streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +81,42 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+    private String childrenToString(List<ProcessorNode<?, ?>> children) {
+        if (children == null || children.isEmpty()) {
+            return "";
+        }
+
+        StringBuilder sb = new StringBuilder("children [");
+        for (ProcessorNode child : children) {
+            sb.append(child.name() + ",");
+        }
+        sb.setLength(sb.length() - 1);
+        sb.append("]\n");
+
+        // recursively print children
+        for (ProcessorNode child : children) {
+            sb.append("\t\t\t" + child.toString());
+            if (!sinkNameToTopic.containsKey(child.name())) {
```

Comment:
I see that you are actually printing the sink nodes. I'm wondering if this if condition is necessary since in line 85 this function will be skipped anyway.